src/wp-includes/webfonts-api/class-wp-webfonts-registry.php

```diff
@@ -0,0 +1,238 @@
+	 *
+	 * @since 5.9.0
+	 *
+	 * @param string $provider_id Provider ID to fetch.
+	 * @return array[] Registered webfonts.
+	 */
+	public function get_by_provider( $provider_id ) {
+		if ( ! isset( $this->registry_by_provider[ $provider_id ] ) ) {
+			return array();
+		}
+
+		$webfonts = array();
+		foreach ( $this->registry_by_provider[ $provider_id ] as $registration_key ) {
+			// Skip if not registered.
+			if ( ! isset( $this->registered[ $registration_key ] ) ) {
+				continue;
+			}
+
+			$webfonts[ $registration_key ] = $this->registered[ $registration_key ];
+		}
+
+		return $webfonts;
+	}
+
+	/**
+	 * Registers the given webfont if its schema is valid.
+	 *
+	 * @since 5.9.0
+	 *
+	 * @param array $webfont {
+	 *     Webfont definition.
+	 *
+	 *    @type string       $provider              The provider ID (e.g. 'local', 'google').
+	 *    @type string       $fontFamily            The @font-face font-family property.
+	 *    @type string       $fontWeight            The @font-face font-weight property.
+	 *                                              The font-weight can be a single value, or a range.
+	 *                                              If a single value, then the font-weight can either be
+	 *                                              a numeric value (400, 700, etc), or a word value (normal, bold, etc).
+	 *                                              If a range, then the font-weight can be a numeric range
+	 *                                              using 2 values, separated by a space ('100 700').
+	 *    @type string       $fontStyle             The @font-face font-style property.
+	 *                                              The font-style can be a valid CSS value (normal, italic etc).
+	 *    @type string       $fontDisplay           The @font-face font-display property.
+	 *                                              Accepted values: 'auto', 'block', 'fallback', 'swap'.
+	 *    @type array|string $src                   The @font-face src property.
+	 *                                              The src can be a single URL, or an array of URLs.
+	 *    @type string       $fontStretch           The @font-face font-stretch property.
+	 *    @type string       $fontVariant           The @font-face font-variant property.
+	 *    @type string       $fontFeatureSettings   The @font-face font-feature-settings property.
+	 *    @type string       $fontVariationSettings The @font-face font-variation-settings property.
+	 *    @type string       $lineHeightOverride    The @font-face line-gap-override property.
+	 *    @type string       $sizeAdjust            The @font-face size-adjust property.
+	 *    @type string       $unicodeRange          The @font-face unicode-range property.
+	 *    @type string       $ascendOverride        The @font-face ascend-override property.
+	 *    @type string       $descendOverride       The @font-face descend-override property.
```

Comment:
Not sure I like that we're using camel-case here. This is not at all a convention for PHP APIs in core, they pretty much all use snake case for values in an array.

I realize this is different particularly in JS, and also in a lot of other PHP projects, but doing this here breaks a convention that has been used by WordPress core since forever. Can we change those to snake case? If we want to later use camel-case in `theme.json`, that's a different story, in there it would totally be expected. We could easily reformat the names accordingly to translate from `theme.json` configuration to the actual API calls here.

This also has some overlap with my other comment in https://github.com/WordPress/wordpress-develop/pull/1736/files#r736980161 - there is no connection to `theme.json` in this PR just yet, so let's build this API similarly to how other core PHP APIs behave.

Reply:
Bringing the conversation over here. 

Sharing this for context:
The primary reason for this is to help extenders who maintain multiple types of themes, i.e. so that configuring a collection of webfonts is similar in schema between `theme.json` and non-`theme.json` themes.

That said, I do see your point about it deviating from how we think about PHP keyed arrays in core itself.

I'll make the updates to:
- the API's documentation
- remove the `kebab-case` converter
- change the testing instruction comments in this PR thread
- change the test data sets

@aristath You'll need to update the test plugins and theme too please.

Reply:
Wait. Don't we normally key arrays in core API's by `snake_case` and not `kebab-case`? So instead of `font-family`, it would be `font_family`. @felixarntz what's your understanding here? If it's in `snake_case`, a converter is needed to convert each property into a valid CSS property, i.e. kebab-case format. Do you think it's okay to specify configuring the keys as valid CSS properties? Or should it be strictly the same as other core APIs to require only `snake_case`?

Reply:
I assume you meant to use kebab-case and not snake_case. Here's the commit https://github.com/WordPress/wordpress-develop/pull/1736/commits/1a8faaafbe2b560eec8a7826be27bc15b0dc7b0f with that change.

Reply:
@hellofromtonya Right, I think we should use `snake_case` for the keys/arguments here, since that is in line with other core PHP APIs. We will still need the converter to `kebab-case` therefore, but instead from `camelCase` it will convert from `snake_case`.

> The primary reason for this is to help extenders who maintain multiple types of themes, i.e. so that configuring a collection of webfonts is similar in schema between `theme.json` and non-`theme.json` themes.

That's a fair point. However, I'd argue that having the PHP API use a different format for names shouldn't be an issue. The format of names can be expected to be different in different contexts, e,g. we also use `post_type` throughout PHP code, and then in some of the Gutenberg JS APIs it is now `postType` - I think that's perfectly fine.

Theme developers that build `theme.json` themes will have to look at the `theme.json` reference either way, and a lot of them won't be familiar with this underlying PHP API. Using `snake_case` here ensures this API uses expected argument names for PHP developers (e.g. plugin developers or current/legacy theme developers), and for `theme.json` we can use `camelCase` to ensure it works as expected for future/full-site editing theme developers.

Reply:
>Right, I think we should use `snake_case` for the keys/arguments here, since that is in line with other core PHP APIs. We will still need the converter to kebab-case therefore, but instead from `camelCase` it will convert from `snake_case`.

Already, I'll update shortly.

Noting for our future selves: `theme.json` will convert from `camelCase` to `snake_case` and then be converted within the API to `kebab-case`.  This shouldn't have an overall impact to performance. However, converting multiple times seems unnecessary. Let's think about that when the time comes.

Reply:
@felixarntz The change is done in this commit https://github.com/WordPress/wordpress-develop/pull/1736/commits/0686a4cc68f966c330e1d4af8965131a081a174a.

The API now requires snake_case properties as an input, e.g. `font_family`. It then converts those keys into valid CSS properties in kebab-case format, e.g. `font-family`.

Note, the API does not prevent kebab-case keys from being passed into it during registration. But it no longer converts camelCase keys into kebab-case. Therefore, the `theme.json` parser will need to convert to either `snake_case` or directly to `kebab-case` when passing its webfonts to the API.

So I think we're good 👍